src/sqlfluff/core/parser/parsers.py

```diff
@@ -100,6 +100,43 @@ def match(
         return MatchResult.from_unmatched(segments)
 
 
+class MultiStringParser(StringParser):
+    """An object which matches and returns raw segments on a collection of strings."""
+
+    def __init__(
+        self,
+        templates: Collection[str],
+        raw_class: Type[RawSegment],
+        name: Optional[str] = None,
+        type: Optional[str] = None,
+        optional: bool = False,
+        **segment_kwargs,
+    ):
+        self.templates = templates
+        self.raw_class = raw_class
+        self.name = name
+        self.type = type
+        self.optional = optional
+        self.segment_kwargs = segment_kwargs or {}
```

Comment:
Should we uppercase during `_init_` to prevent having to do it for each match?

Also should this just call super like RegexParse does for the other values? In case the base StringParser ever changes in future?

Reply:
> including setting template to an empty string?
This is what feels odd to me. I'm not sure we should set an attribute which isn't relevant to the class at all.  

> I wonder should StringParser also be changed to upper at __init__ time?
Quite possibly, though I suspect the overhead is very small!

Reply:
> This is what feels odd to me. I'm not sure we should set an attribute which isn't relevant to the class at all.

Yeah I get your hesitancy. Let's leave this comment open and let the others weigh in.